### PR TITLE
Issues zonemaster engine 577 and 582

### PR DESCRIPTION
**Pull request by vlevigneron**
Fixes consistency05 issues.
Fixes zonemaster/zonemaster-engine#577
Fixes zonemaster/zonemaster-engine#582.

### CODE REVIEW

File: lib/Zonemaster/Engine/Test/Consistency.pm [hunk 1]
```diff
@@ -179,6 +191,10 @@ Readonly my %TAG_DESCRIPTIONS => (
     ONE_SOA_TIME_PARAMETER_SET => sub {         # ONE_SOA_TIME_PARAMETER_SET
         __x 'A single SOA time parameter set was seen (REFRESH={refresh},RETRY={retry},EXPIRE={expire},MINIMUM={minimum}).', @_;
     },
+    OUT_OF_BAILIWICK_ADDR_MISMATCH => sub {
+        __x    # OUT_OF_BAILIWICK_ADDR_MISMATCH
+          'Out of Bailiwick addresses mismatch between parent ({parent_addresses}) and child ({zone_addresses})', @_;
```
[the remark this answers is not in the document]
Comment (vlevigneron) [reply lib/Zonemaster/Engine/Test/Consistency.pm]: Fixed in https://github.com/zonemaster/zonemaster-engine/pull/593/commits/a169faa792a3533d9e226ab66a0e25e65b5cf9a4

File: lib/Zonemaster/Engine/Test/Consistency.pm [hunk 2]
```diff
@@ -125,12 +125,24 @@ Readonly my %TAG_DESCRIPTIONS => (
...
+    },
     EXTRA_ADDRESS_CHILD => sub {    # EXTRA_ADDRESS_CHILD
         __x 'Child has extra nameserver IP address(es) not listed at parent ({addresses}).', @_;
     },
     EXTRA_ADDRESS_PARENT => sub {    # EXTRA_ADDRESS_PARENT
         __x 'Parent has extra nameserver IP address(es) not listed at child ({addresses}).', @_;
     },
+    IN_BAILIWICK_ADDR_MISMATCH => sub {
+        __x    # IN_BAILIWICK_ADDR_MISMATCH
+          'In Bailiwick addresses mismatch between parent ({parent_addresses}) and child ({zone_addresses})', @_;
```
[the remark this answers is not in the document]
Comment (vlevigneron) [reply lib/Zonemaster/Engine/Test/Consistency.pm]: Fixed in https://github.com/zonemaster/zonemaster-engine/pull/593/commits/a169faa792a3533d9e226ab66a0e25e65b5cf9a4

File: lib/Zonemaster/Engine/Test/Consistency.pm [hunk 3]
```diff
@@ -662,13 +678,41 @@ sub consistency05 {
 
     my %strict_glue;
     my %extended_glue;
-    for my $ns ( @{ Zonemaster::Engine::TestMethods->method4( $zone ) } ) {
-        my $ns_string = $ns->name->fqdn . "/" . $ns->address->short;
-        if ( $zone->name->is_in_bailiwick( $ns->name ) ) {
+
+    # We need to work on Methods...
```
[the remark this answers is not in the document]
Comment (vlevigneron) [reply lib/Zonemaster/Engine/Test/Consistency.pm]: Yes it is planned for the end of this year.

File: lib/Zonemaster/Engine/Test/Consistency.pm [hunk 4]
```diff
@@ -126,6 +126,14 @@ Readonly my %TAG_DESCRIPTIONS => (
         __x    # ADDRESSES_MATCH
           'Glue records are consistent between glue and authoritative data.', @_;
     },
+    CHILD_NS_FAILED => sub {
+        __x    # CHILD_NS_FAILED
+          'Query to {ns}/{address} failed', @_;
```
Comment (matsduf) [inline lib/Zonemaster/Engine/Test/Consistency.pm]:
```
'Unexepected or erroneous reply from {ns}/{address}.', @_;
```

Comment (vlevigneron) [reply lib/Zonemaster/Engine/Test/Consistency.pm]: Fixed in https://github.com/zonemaster/zonemaster-engine/pull/593/commits/a1148f2070cb24def1fa4ed8b2aef6a5d6c6c4f8

**Pull request merged**